ms_active_directory/environment/ldap/ldap_format_utils.py

```diff
@@ -368,3 +368,11 @@ def validate_and_normalize_logon_name(name: str, supports_legacy_behavior: bool)
             raise InvalidDomainParameterException('Common names may not contain any of the following characters: '
                                                   '{}'.format(', '.join(AD_USERNAME_RESTRICTED_CHARS)))
     return name
+
+
+def construct_primary_group_sid(object_sid: str, primary_group_id: int) -> str:
+    """ Construct the objectSid for the primary group of an object given the objectSid of the object in question
+    and the primaryGroupID for the object.
+    """
+    base_sid = object_sid[:object_sid.rfind('-')]
```

Comment:
isn't this just the domain SID? if so, could we just pass in the domain SID rather than pull the user's SID (which may be more privileged/protected)? the domain SID gets cached within a session so it won't adversely impact performance

Reply:
Yes, this is the domain SID. I agree this would be a better way of doing this so we don't have to pull the base SID from the user. Are you saying that the domain SID is already getting cached in a session? Or is this something that needs to be added?

Reply:
huh i thought i added it at the same time I added `find_netbios_name` but it seems like i might have never pushed it. I just checked and it's still on an (out of date) local branch.

we can leave this as is for now and put a TODO to switch over once I push those changes up

Reply:
Added the TODO.